torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,427 @@
+        """
+        download_url(
+            url,
+            self.root,
+            filename=os.path.split(url)[1],
+            md5=md5 if self.checksum else None,
+        )
+
+    def _extract(self, path: str) -> None:
+        """Extracts a modality.
+
+        Args:
+            path: path to the modality folder
+        """
+        filepath = os.path.join(self.root, os.path.split(path)[1])
+        extract_archive(filepath)
+
+    def _merge_parts(self, modality: str) -> None:
+        """Merge the modalities that are downloaded and extracted in two parts.
+
+        Args:
+            root: root directory where dataset can be found
+            modality: the filename of the modality
+        """
+        # Create a new folder named after the 'modality' variable
+        modality_folder = os.path.join(self.root, modality)
+        os.makedirs(
+            modality_folder, exist_ok=True
+        )  # Will not raise an error if the folder already exists
+
+        # List of source folders
+        source_folders = [
+            os.path.join(self.root, modality + "_part1"),
+            os.path.join(self.root, modality + "_part2"),
+        ]
+
+        # Move files from each source folder to the new 'modality' folder
+        for source_folder in source_folders:
+            for file_name in os.listdir(source_folder):
+                source = os.path.join(source_folder, file_name)
+                destination = os.path.join(modality_folder, file_name)
+                if os.path.isfile(source):
+                    shutil.copy(source, destination)  # Move files to 'modality' folder
+
+    def _convert_to_color(
+        self, arr_2d: Tensor, cmap: dict[int, tuple[int, int, int]]
+    ) -> "np.typing.NDArray[np.uint8]":
+        """Numeric labels to RGB-color encoding.
+
+        Args:
+            arr_2d: 2D array to be colorized
+            cmap: colormap to use when mapping the labels
+
+        Returns:
+            3D colored image
+        """
+        arr_3d = np.zeros((arr_2d.shape[0], arr_2d.shape[1], 3), dtype=np.uint8)
+
+        for c, i in cmap.items():
+            m = arr_2d == c
+            arr_3d[m] = i
+        return arr_3d
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample image-mask pair returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        modality_channels = {"viirs": 1, "esa_wc": 1, "s1": 2}
+
+        # Assign a number of channels to all 's2' modalities
+        for modality in self.modality:
+            if modality.startswith("s2"):
+                modality_channels[modality] = 10
```

Comment:
If you just want this to be 10 for all modes not listed, you could use a `defaultdict(10)`